src/App.js

```diff
@@ -1,101 +1,129 @@
+      isComplete: timerToUpdate.currentRep + 1 >= timerToUpdate.targetRep, // since you are updating the currentRep & isComplete at the same time, you have to artificially add one to the currentRep for this comparison. This is another reason to remove `isComplete` from state.
+    }
+    newTimers[timerToUpdateIndex] = updatedTimer
+    setTimers(newTimers)
   }
-  
-  const timersList = timers.map(item => {
-    return (
-      <Timer 
-        key={item.id}
-        {...item}
-        changeCompletion={ changeCompletion }
-      />
-    )
-  }).reverse()
 
-  function addTimer(newTimer) {
-    setTimers(prevTimer => ([
+  function addTimer_old(newTimer) {
+    // you're making a shallow copy of the timers here, so the references to the individual timers are staying the same. You should make a deep copy instead.
+    setTimers((prevTimer) => [
       ...prevTimer,
       {
         ...newTimer,
-        id: prevTimer.length + 1
-      }
-    ]))
+        id: prevTimer.length + 1, // relying on length is bug-prone because you could add a timer (id = 4), then delete a different time and add another and id would equal 4 again.
+      },
+    ])
   }
 
-  function handleDelete(event) {
-    console.log(timers.id)
+  function addTimer(newTimer) {
+    const newTimers = timers.map((timer) => ({ ...timer })) // make a deep copy of an array of objects so you don't accidentally update state directly
+    const nextId = Math.max(...newTimers.map((timer) => timer.id)) + 1 // this is safer than just relying on length. Get the max of the ids, then add one. Even better would to use UUIDs but that may add complexity where it isn't needed yet.
+    newTimers.push({
+      ...newTimer,
+      id: nextId,
+    })
+    setTimers(newTimers)
+  }
+
+  function handleDelete_old(event) {
+    console.log(timers.id) // console.logs are a great way to debug, just remember to remove them once you've used them.
     console.log(event.target.id)
-    setTimers(prevTimers => prevTimers.filter(prevTimers => prevTimers.id !== parseInt(event.target.id))) // != didn't go trough build
-}
+    setTimers((prevTimers) =>
+      prevTimers.filter((prevTimers) => prevTimers.id !== parseInt(event.target.id))
+    ) // != didn't go trough build
+    // yeah Ideally your IDs would be strings, not numbers, so you wouldn't have to go around parsing them from events and such
+  }
+
+  function handleDelete(event) {
+    const newTimers = timers.filter((timer) => timer.id !== parseInt(event.target.id))
+    setTimers(newTimers)
+  }
 
   return (
-    <div className="App">
+    <div className='App'>
       <BrowserRouter>
-      <Navbar />
+        <Navbar />
         <Routes>
-          <Route path="/" element={ timersList } />
-          <Route 
-            path="manage" 
-            element={
-              <ManageTimers 
-                timers={ timers } 
-                handleDelete={ handleDelete }
-              />}
+          <Route
```

Comment:
App file should only contain Route file, create another file Routes where all the Routes are mentioned